msteams-platform/bots/how-to/conversations/command-bot-in-teams.md

```diff
@@ -0,0 +1,335 @@
+
+      export class xxxCommandHandler implements TeamsFxBotCommandHandler {
+          triggerPatterns: TriggerPatterns = "<string or RegExp pattern to trigger the command>";
+
+          async handleCommandReceived(
+             context: TurnContext,
+             message: CommandMessage
+          ): Promise<string | Partial<Activity>> {
+             // verify the command arguments which are received from the client if needed.
+             console.log(`Bot received message: ${message.text}`);
+
+             // do something to process your command and return message activity as the response.
+             // You can leverage `MessageBuilder` utilities from the `@microsoft/teamsfx` SDK 
+             // to facilitate building message with cards supported in Teams.
+         }    
+       }
+
+   ```
+
+1. Provide the `triggerPatterns` that can trigger the command handler. It's the command name defined in your manifest, or you can use `RegExp` to handle a complex command such as with some options in the command message.
+
+1. You can implement `handleCommandReceived` to handle the command and return a response that is used to notify the users:
+
+   * If needed you can retrieve useful information for the conversation from the `context` parameter.
+
+   * If needed parse command input:
+
+     * `message.text`: Use the input message.
+     * `message.matches`: Use the capture groups, if you use the `RegExp` for `triggerPatterns` to trigger the command.
+
+<br>
+
+</details>
+
+<details>
+
+<summary><b>4. Register the new command</b></summary>
+
+Open `bot\src\internal\initialize.ts` and update the call to `ConversationBot` constructor to include your newly added command handlers.
+
+ ```TypeScript
+       export const commandBot = new ConversationBot({
+           // The bot id and password to create BotFrameworkAdapter.
+           // See https://aka.ms/about-bot-adapter to learn more about adapters.
+           adapterConfig: {
+              appId: process.env.BOT_ID,
+              appPassword: process.env.BOT_PASSWORD,
+           },
+           command: {
+               enabled: true,
+               commands: [ new HelloWorldCommandHandler(), new xxxCommandHandler() ],
+           },
+      });
+ ```
+
+Additionally, you can use `registerCommand`, or `registerCommands` API from your `ConversationBot` instance to incrementally add your command(s) to a command bot.
+
+```
+    // register a single command
+    commandBot.command.registerCommand(new xxxCommandHandler());
+
+    // register a set of commands
+    commandBot.command.registerCommands([
+        new xxxCommandHandler(),
+        new yyyCommandHandler()
+    ]);
+```
+
+By completing the steps of adding a new command and response into your bot app, you can press F5 to debug locally with the command-response bot. Otherwise you can provision and deploy commands to deploy the change to Azure.
+<br>
+
+</details>
+
+## Extend command bot with other bot scenarios
+
```

Comment:
This entire section is duplicate with FAQs - How to extend... can we consolidate?

Reply:
Do you want me to remove the FAQ?

Reply:
I'm suggesting to leave only one section. Considering notification bot / workflow bot has similar content for extending with each other and that are in FAQ, we can leave those contents in FAQ I think.